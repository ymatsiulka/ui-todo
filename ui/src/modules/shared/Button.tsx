import React from 'react';
import classNames from 'classnames';
import { ButtonProps } from 'types/frontend';
import styles from './Button.m.scss';

const Button: React.FC<ButtonProps> = ({ className, children, type, title, ...props }) => {
  const buttonClassNames = classNames(styles.button, className);

  return (
    <button {...props} className={buttonClassNames} type={type ?? 'button'}>
      <span>{children ?? title}</span>
    </button>
  );
};

export default Button;
