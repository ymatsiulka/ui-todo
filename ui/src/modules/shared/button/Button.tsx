import React from 'react';
import classNames from 'classnames';
import { ButtonProps } from 'types/frontend';

const Button: React.FC<ButtonProps> = ({ className, children, type, title, ...props }) => {
  const buttonClassNames = classNames('button', className);

  return (
    <button {...props} className={buttonClassNames} type={type}>
      <span>{children ?? title}</span>
    </button>
  );
};

export default Button;
