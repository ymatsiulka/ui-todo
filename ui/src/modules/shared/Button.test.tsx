import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

test('Button should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(
    <Button className="btn-primary" type="button" title="primary">
      Some text
    </Button>,
  );

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
