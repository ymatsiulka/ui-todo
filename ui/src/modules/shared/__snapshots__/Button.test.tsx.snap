// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<DocumentFragment>
  <button
    class="button btn-primary"
    type="button"
  >
    <span>
      Some text
    </span>
  </button>
</DocumentFragment>
`;
